### PR TITLE
Issue #2052 - Change reachability computation

### PR DESCRIPTION
**Pull request by MarcoSantarossa**
It fixes the issue #2052. There is a bug with the reachability computation when the user has a sim but s/he didn't select the carrier yet. In this scenario we have both `.connectionRequired` and `.isWWAN`.

### CODE REVIEW

File: Source/NetworkReachabilityManager.swift [hunk 1]
```diff
@@ -184,20 +184,24 @@ public class NetworkReachabilityManager {
...
         #if os(iOS)
             if flags.contains(.isWWAN) { networkStatus = .reachable(.wwan) }
         #endif
 
         return networkStatus
     }
+    
+    private func isNetworkReachable(with flags: SCNetworkReachabilityFlags) -> Bool {
+        let needsConnection = flags.contains(.connectionRequired)
+        let canConnectionAutomatically = flags.contains(.connectionOnDemand) || flags.contains(.connectionOnTraffic)
```
Comment (macbellingrath) [inline Source/NetworkReachabilityManager.swift]: could this name be canConnectAutomatically ?

File: Source/NetworkReachabilityManager.swift [hunk 2]
```diff
@@ -184,20 +184,24 @@ public class NetworkReachabilityManager {
...
         guard flags.contains(.reachable) else { return .notReachable }
 
-        var networkStatus: NetworkReachabilityStatus = .notReachable
-
-        if !flags.contains(.connectionRequired) { networkStatus = .reachable(.ethernetOrWiFi) }
-
-        if flags.contains(.connectionOnDemand) || flags.contains(.connectionOnTraffic) {
-            if !flags.contains(.interventionRequired) { networkStatus = .reachable(.ethernetOrWiFi) }
-        }
+        let isNetworkReachable = self.isNetworkReachable(with: flags)
```
Comment (macbellingrath) [inline Source/NetworkReachabilityManager.swift]: could you combine this with the guard statement above?

into:
```swift
 guard flags.contains(.reachable), isNetworkReachable(with:flags) else { return .notReachable }
```

**Comment by MarcoSantarossa** [comment]
I think CI for watchOS has some issues

**Comment by MarcoSantarossa** [comment]
@jshier can you rerun the tests please? They pass locally so it seems an odd failure.

**Comment by jshier** [comment]
No problem, it's a Travis issue (still).

**Comment by MarcoSantarossa** [comment]
@jshier is this PR under review? Or Don't you want it merged?

**Comment by jshier** [comment]
@MarcoSantarossa We're still reviewing changes for our 4.5 release, we'll likely review and merge soon.

**Comment by cnoon** [comment]
Thanks for putting this together @MarcoSantarossa...much appreciated! 🍻 I've made a few small tweaks to your PR and have pushed it into `master` in 68f4eb3f while maintaining your attribution. This change will go out shortly in the AF 4.5 release.

Thanks again!

**Pull request closed without merging**